modules/Bio/EnsEMBL/Hive/AnalysisStats.pm

```diff
@@ -249,31 +265,99 @@ sub get_or_estimate_batch_size {
 }
 
 
-sub estimate_num_required_workers {     # this 'max allowed' total includes the ones that are currently running
+sub estimate_num_required_workers {     # this doesn't count the workers that are currently running
     my $self                = shift @_;
     my $remaining_job_count = shift @_ || 0;    # FIXME: a better estimate would be $self->claimed_job_count when it is introduced
 
-    my $num_required_workers = $self->ready_job_count + $remaining_job_count;   # this 'max' estimation can still be zero
+    # Guaranteed to be non-zero
+    my $avg_msec_per_job = $self->get_or_estimate_avg_msec_per_job;
+    # Number of jobs earmarked for each worker (cannot be zero)
+    my $num_jobs_per_worker = POSIX::ceil($self->min_worker_runtime / $avg_msec_per_job );
+
+    # Work left to do
+    my $jobs_to_do = $self->ready_job_count + $remaining_job_count;
+
+    # Correction so that 1) each worker runs for a minimum amount of time
+    # and 1) existing workers get a chance to do some more work besides
+    # their current job.
+    # NOTE: the /2 factor is the equivalent of the TailTrimming mechanism
+    # NOTE: if the analysis has never been run, it will assign
+    #       min_worker_runtime/min_job_runtime_msec=12,000 jobs to each worker
+    my $num_required_workers = POSIX::ceil( ($jobs_to_do + $self->num_running_workers) / $num_jobs_per_worker - $self->num_running_workers/2 );
```

Comment:
This is not the right formula yet. It doesn't obey the constraints:

* `$jobs_to_do` == 0 -> `$num_required_workers` = 0 (regardless of `$self->num_running_workers`), i.e. stop submitting workers when there is nothing left to do
* `$num_required_workers` <= `$jobs_to_do`, i.e don't submit more workers than there are jobs to do